app/services/activitypub/verify_quote_service.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class ActivityPub::VerifyQuoteService < BaseService
+  include JsonLdHelper
+
+  # Optionally fetch quoted post, and verify the quote is authorized
+  def call(quote, fetchable_quoted_uri: nil, prefetched_body: nil, request_id: nil)
+    @request_id = request_id
+    @quote = quote
+    @fetching_error = nil
+
+    fetch_quoted_post_if_needed!(fetchable_quoted_uri)
+    return if fast_track_approval! || quote.approval_uri.blank?
+
+    @json = fetch_approval_object(quote.approval_uri, prefetched_body:)
+    if @json.nil?
+      # It is possible the proof isn't dereferenceable yet, so only reject if already accepted
+      quote.reject! unless @quote.pending?
+      return
+    end
+
+    return if non_matching_uri_hosts?(quote.approval_uri, value_or_id(@json['attributedTo']))
+    return unless matching_type? && matching_quote_uri?
+
+    # Opportunistically import embedded posts if needed
```

Comment:
When is this needed?

Could this be skipped for posts just fetched in line 12 above?

Reply:
It could be if dereferencing the post is not possible for some reason (e.g. at some point Pleroma made a point of never replying to requests for private posts, even in response to a correctly signed request with on behalf of an account who has access to it; not sure if it changed), but the object is bundled into the activity (e.g. when receiving it directly); it's kind of an edge case, though, and would be even less relevant if we move towards an attribute that isn't a `xsd:anyURI` for handling quotes (such as the proposed `quote`)

It should be skipped for posts jut fetched in line 12 above thanks to the `@quote.quoted_status_id.present?` check.